### PR TITLE
added guards against uris resolving in different ways

### PR DESCRIPTION
**Pull request by fuag15**

### CODE REVIEW

File: src/aardvark/aardvark_scene_graph.cpp [hunk 1]
```diff
@@ -80,6 +80,7 @@ namespace aardvark
 	void from_json( const nlohmann::json& j, GadgetParams_t& params )
 	{
 		j.at( "uri" ).get_to( params.uri );
+		if(params.uri.back() == '/') params.uri.pop_back();
```
Comment (JoeLudwig) [inline src/aardvark/aardvark_scene_graph.cpp:83]: Maybe add the same function on the C++ side? There are likely other places that will need this. (Or already have it.)

**Review by JoeLudwig — CHANGES_REQUESTED** [review]

File: src/aardvark/aardvark_scene_graph.cpp (hunk 1 above)
Comment (fuag15) [reply src/aardvark/aardvark_scene_graph.cpp:83]: Found the origin of the bad data. Looks like this function doesn't resolve to the correct format (at least in firefox / chrome) https://github.com/fuag15/aardvark/blob/e18092c552437bc817a2cc0794de9b34ff20ec1b/packages/aardvark-react/src/aardvark_gadget.tsx#L70-L82 What's your preference with fixing this function vs adding some data validation in the c++ side

Comment (fuag15) [reply src/aardvark/aardvark_scene_graph.cpp]: Moved it to a helper func and changed the source of the data to send the uri without a / at the end debated putting an initializer on that struct but didn't see that pattern around so opted for a free func in the aardvark namespace

**Review by JoeLudwig — APPROVED** [review]

File: packages/aardvark-react/src/aardvark_gadget.tsx [hunk 2]
```diff
@@ -69,16 +69,8 @@ interface ActionStateListener
...
-		return window.location.origin
-			+ window.location.pathname.slice( 0, window.location.pathname.lastIndexOf( "/" ) );
-		//console.log( "Stripping gadget URI down to", this.m_actualGadgetUri );
-	}
-	else
-	{
-		return window.location.origin + window.location.pathname;
-	}
+	return window.location.origin
+		+ window.location.pathname.slice( 0, window.location.pathname.lastIndexOf( "/" ) );
```
[the remark this answers is not in the document]
Comment (fuag15) [reply packages/aardvark-react/src/aardvark_gadget.tsx:73]: afaik pathname always at least has a single '/' in it

**Pull request merged**